CCPAPI/sip/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+    message(SEND_ERROR "no Python found matching ${PYTHON_PREFERED_VERSION}.x")
+else()
+    message(STATUS "Python found ${PYTHON_VERSION_STRING} matches prefered version ${PYTHON_PREFERED_VERSION}.x")
+    message(STATUS "PYTHON_LIBRARIES:" ${PYTHON_LIBRARIES})
+endif()
+
+# --- PyQt Sip detection
+
+find_package(PYQTSIP REQUIRED)  # should come after Python
+
+# --- ----------------------------------------------------------
+# --- target PyQt sip Python library: Python module cloudCompare
+
+project( cloudCompare VERSION 0.1 )
+
+add_library( cloudCompare SHARED "")
+
+# --- sources / sip wrappings
+
+set(SIP_MAIN_SOURCE cloudCompare.sip)
+set(SIP_ADD_SOURCES
+    ccPApi.sip
+    ccPointCloud.sip
+    ScalarField.sip
+    )
+set(SIP_ALL_SOURCES ${SIP_MAIN_SOURCE} ${SIP_ADD_SOURCES})
+message(STATUS "SIP_ALL_SOURCES: " "${SIP_ALL_SOURCES}")
+
+# TODO: names of generated sip files follow some rules based on module name, class name, namespace etc. Not obvious to compute...
+
+set(SIP_MAIN_GENSRC ${CMAKE_CURRENT_BINARY_DIR}/sipcloudComparecmodule.cc)
+set(SIP_ADD_GENSRC
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareccPointCloud.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareQList0101ccPointCloud.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareCCLib.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareCCLibScalarField.cc
+   )
+set(SIP_ALL_GENSRC ${SIP_MAIN_GENSRC} ${SIP_ADD_GENSRC})
+
+message(STATUS "SIP_ALL_GENSRC: " "${SIP_ALL_GENSRC}")
+
+set_source_files_properties( ${SIP_ALL_GENSRC} PROPERTIES GENERATED TRUE )
+
+set(SIP_OPTIONS -s .cc -c . ${PYQT_SIPFLAGS})
+message(STATUS "SIP_OPTIONS: " "${SIP_OPTIONS}")
+
+add_custom_command(
+    OUTPUT ${SIP_ALL_GENSRC}
+    COMMAND ${SIP_EXECUTABLE} ${SIP_OPTIONS} ${CMAKE_CURRENT_SOURCE_DIR}/${SIP_MAIN_SOURCE}
+    DEPENDS ${SIP_ALL_SOURCES}
+    )
+
+# --- sources / to compile
+
+target_sources( cloudCompare
+                PRIVATE
+                ${SIP_ALL_GENSRC}
+                )
+                
+# --- Python wrapping with sip
+
+target_include_directories( cloudCompare PUBLIC
+    ${SIP_INCLUDE_DIR}
+    ${PYTHON_INCLUDE_DIRS}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    )
+
+# --- Numpy includes
+
+find_path(NUMPY_INCLUDE_DIR numpy/npy_common.h HINTS ${PYTHON_INCLUDE_DIRS} DOC "directory path of include: numpy/npy_common.h")
+message(STATUS "NUMPY_INCLUDE_DIR: " ${NUMPY_INCLUDE_DIR})
+
+target_include_directories( cloudCompare PUBLIC
+    ${NUMPY_INCLUDE_DIR}
+    )
+
+target_link_libraries( cloudCompare
+    CC_PAPI_LIB
+    CCCoreLib
+    CC_FBO_LIB
+    QCC_DB_LIB
+    QCC_IO_LIB 
+    QCC_GL_LIB
+    qcustomplot
+    ${PYTHON_LIBRARIES}
+    )
+
+# Qt libraries
+
+target_link_libraries( cloudCompare
+    Qt5::Core
+    Qt5::Gui
+    Qt5::Widgets
+    Qt5::OpenGL
+    Qt5::PrintSupport
+    )
+
+if (WIN32)
+   target_link_libraries( cloudCompare Qt5::WinMain)
+endif()
+
+if( WIN32 OR APPLE)
+   foreach( dest ${INSTALL_DESTINATIONS} )
```

Comment:
I'm unclear how this is supposed to work.

This is generating a library and right now it is installed into the CloudCompare and the ccViewer install directories.

Then you use it from Python somehow?